File: batch_rename_files.m
function batch_rename_files(datadir, searchstr, removestr, replacementstr)
% Rename all files matching string pattern inside a directory. 
% Note that this can only replace the filename, not other parts of the path
% or the file extension.
% npm 2019

% Retrieve all the matching file names
files = dir(fullfile(datadir, searchstr))

for i=1:length(files)
    % split up the filename
    [pathn, filename, extension] = fileparts(files(i).name)
    % Create the new name
    newfn = strrep(filename, removestr, replacementstr)
    movefile([pathn filename extension], [pathn newfn extension])
end


return

File: batch_rename_files.py
import os
import glob

# Directory and search pattern
datadir = '/path/to/data/'
searchstr = 'data*Time_0*_Angle_*.ome.tif'
removestr = 'data\\'   # note: backslash must be escaped
replacementstr = ''

# Get matching files
files = glob.glob(os.path.join(datadir, searchstr))

# Iterate through files and rename
for oldfn in files:
    pathn, filename = os.path.split(oldfn)
    newfilename = filename.replace(removestr, replacementstr)
    newfn = os.path.join(pathn, newfilename)
    
    print(f"Replacing {filename}\n  with -> {newfilename}")
    os.rename(oldfn, newfn)
